File: CarSpecAngular/carspec/src/app/core/models/models/variantDto.ts
export class CreateVariantDto {
  variantName: string;
  engineId: number;
  transmissionId: number;
  drivetrainId: number;
  exShowroomPrice: number;
  variantImageUrl: string | null;

  constructor(
    variantName: string,
    engineId: number,
    transmissionId: number,
    drivetrainId: number,
    exShowroomPrice: number,
    variantImageUrl: string | null = null
  ) {
    this.variantName = variantName;
    this.engineId = engineId;
    this.transmissionId = transmissionId;
    this.drivetrainId = drivetrainId;
    this.exShowroomPrice = exShowroomPrice;
    this.variantImageUrl = variantImageUrl;
  }
}

export class VariantDto {
  variantId: number;
  variantName: string;
  fuelType: string;
  cubicCapacity: number;
  isTurbocharged: boolean;
  transmissionType: string;
  maxPower: number;
  maxTorque: number;
  exShowroomPrice: number;

  constructor(
    variantId: number,
    variantName: string,
    fuelType: string,
    cubicCapacity: number,
    isTurbocharged: boolean,
    transmissionType: string,
    maxPower: number,
    maxTorque: number,
    exShowroomPrice: number
  ) {
    this.variantId = variantId;
    this.variantName = variantName;
    this.fuelType = fuelType;
    this.cubicCapacity = cubicCapacity;
    this.isTurbocharged = isTurbocharged;
    this.transmissionType = transmissionType;
    this.maxPower = maxPower;
    this.maxTorque = maxTorque;
    this.exShowroomPrice = exShowroomPrice;
  }
}

File: CarSpecAPI/Data/Models/ResponseModel/VariantDto.cs
namespace CarSpecAPI.Data.Models.ResponseModel
{
    public class VariantDto
    {
        public int VariantId { get; set; }

        public string VariantName { get; set; } = string.Empty;

        public string FuelType { get; set; } = string.Empty;

        public decimal? CubicCapacity { get; set; }

        public bool isTurbocharged { get; set; } = false;

        public string TransmissionType {  get; set; } = string.Empty;

        public decimal? MaxPower { get; set; }

        public decimal? MaxTorque { get; set; }

        public decimal? ExShowroomPrice { get; set; }
    }
}
